### PR TITLE
fix: Support `untyped raw ptr`

### PR DESCRIPTION
**Pull request by segfault-magnet**
fixes: #600 

Since the `untyped raw ptr` is not directly used in the SDK we'll skip its generation and avoid delving deeper into `Vec` when encountered in type resolving logic such as `resolve_type` or `ParamType::try_from_type_declaration`.

### CODE REVIEW

**Comment by mohammadfawaz** [comment]
Thank you for this @segfault-magnet! Is `Vec` working again in the ABI with this change?

**Comment by segfault-magnet** [comment]
> Thank you for this @segfault-magnet! Is `Vec` working again in the ABI with this change?

vector e2e tests are passing. They are relatively extensive so I'm pretty sure we're in the clear.

Vector returns are still not supported, but I guess that goes without saying.

**Comment by mohammadfawaz** [comment]
> 

Excellent, looking forward to the next release then 😄  Thank you for addressing this so quickly!

File: packages/fuels/tests/types/generics/src/main.sw [hunk 1]
```diff
@@ -25,7 +25,7 @@ enum EnumWGeneric<N> {
 
 struct MegaExample<T, U> {
     a: ([U; 2], T),
-    b: ([EnumWGeneric<StructWTupleGeneric<StructWArrayGeneric<PassTheGenericOn<T>>>>; 1], u32),
+    b: Vec<([EnumWGeneric<StructWTupleGeneric<StructWArrayGeneric<PassTheGenericOn<T>>>>; 1], u32)>,
```
Comment (digorithm) [inline packages/fuels/tests/types/generics/src/main.sw:28]: 😆

**Review by digorithm — APPROVED** [review]
Awesome work; left a tiny nit but feel free to ignore/push back.

File: packages/fuels/tests/types/generics/src/main.sw (hunk 1 above)
Comment (segfault-magnet) [reply packages/fuels/tests/types/generics/src/main.sw:28]: A PITA to maintain, but I do feel safe seeing it pass.

**Review by digorithm — APPROVED** [review]
:shipit:

**Pull request merged**